src/NuGet.Clients/NuGet.CommandLine/Program.cs

```diff
@@ -334,5 +424,60 @@ private static void SetConsoleInteractivity(IConsole console, Command command)
                 console.Verbosity = command.Verbosity;
             }
         }
+
+        /// <summary>
+        /// Splits up a string of command line arguments into an array
+        /// </summary>
+        /// <param name="unsplitArguments">The string to split</param>
+        /// <returns>An array with split command line arguments</returns>
+        private static string[] SplitArgs(string unsplitArguments)
+        {
+            if (string.IsNullOrWhiteSpace(unsplitArguments))
+            {
+                return new string[0];
+            }
+
+            IntPtr ptrToSplitArgs = CommandLineToArgvW(unsplitArguments, out int numberOfArgs);
+            if (ptrToSplitArgs == IntPtr.Zero)
+            {
+                return new string[0];
+            }
+
+            try
+            {
+                string[] splitArgs = new string[numberOfArgs];
+                for (int i = 0; i < numberOfArgs; i++)
+                {
+                    splitArgs[i] = Marshal.PtrToStringUni(Marshal.ReadIntPtr(ptrToSplitArgs, i * IntPtr.Size));
+                }
+
+                return splitArgs;
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(ptrToSplitArgs);
+            }
+        }
+
+        /// <summary>
+        /// Parses a Unicode command line string and returns an array of pointers to the command line arguments, 
+        /// along with a count of such arguments, in a way that is similar to the standard C run-time argv and argc values.
+        /// </summary>
+        /// <param name="commandLine">Unicode string that contains the full command line.</param>
+        /// <param name="argsLength">Pointer to an int that receives the number of array elements returned</param>
+        /// <returns></returns>
+        [DllImport("shell32.dll", SetLastError = true)]
+        static extern IntPtr CommandLineToArgvW([MarshalAs(UnmanagedType.LPWStr)] string commandLine, out int argsLength);
```

Comment:
What is the benefit of using this over doing a normal string split? It seems like this will make things difficult for mono/cross platform scenario. I would prefer simple/consistent handling for this across all platforms.

Reply:
Nearly all windows command line parsing goes thru this function (including .net). We should not reinvent the wheel and have to deal with a lot of unstable code.

Reply:
Unfortunately mono is a requirement for nuget, I don't think this library is an option.

Reply:
nuget.exe ships as part of mono